File: android/src/main/java/expo/modules/khaltipaymentsdk/exceptions/PaymentExceptions.kt
package expo.modules.khalticheckout.exceptions

import expo.modules.kotlin.exception.CodedException

/**
 * Custom exceptions for Khalti Payment SDK error handling.
 */

class InvalidConfigException(message: String) : CodedException("ERR_INVALID_CONFIG", message, null)

class NoActivityException : CodedException("ERR_NO_ACTIVITY", "No current activity available", null)

class PaymentInitException(message: String) : CodedException("ERR_PAYMENT_INIT", message, null)

class PaymentFailedException(message: String, cause: Throwable? = null) : CodedException("ERR_PAYMENT_FAILED", message, cause)

class PaymentCanceledException : CodedException("ERR_PAYMENT_CANCELED", "Payment was cancelled by the user", null)

File: ios/PaymentExceptions.swift
import ExpoModulesCore
import Foundation

/// Custom exceptions for Khalti Payment SDK
class PaymentException: Exception {
    override var reason: String {
        return "Payment error occurred"
    }
}

class PaymentInitException: Exception {
    override var reason: String {
        return "Failed to initialize payment"
    }

    override var code: String {
        return KhaltiConstants.errorPaymentInit
    }
}

class PaymentFailedException: Exception {
    override var reason: String {
        return "Payment failed"
    }

    override var code: String {
        return KhaltiConstants.errorPaymentFailed
    }
}

class PaymentCanceledException: Exception {
    override var reason: String {
        return "Payment was cancelled by user"
    }

    override var code: String {
        return KhaltiConstants.errorPaymentCanceled
    }
}

class NoViewControllerException: Exception {
    override var reason: String {
        return "No view controller available for presentation"
    }

    override var code: String {
        return KhaltiConstants.errorNoViewController
    }
}

class InvalidConfigException: Exception {
    override var reason: String {
        return "Invalid payment configuration"
    }

    override var code: String {
        return KhaltiConstants.errorInvalidConfig
    }
}
